types/jest/index.d.ts

```diff
@@ -28,1557 +28,22 @@
-         */
-        allArgs(): any[];
-        /**
-         * By chaining the spy with calls.all(), will return the
-         * context (the this) and arguments passed all calls
-         */
-        all(): CallInfo[];
-        /**
-         * By chaining the spy with calls.mostRecent(), will return the
-         * context (the this) and arguments for the most recent call
-         */
-        mostRecent(): CallInfo;
-        /**
-         * By chaining the spy with calls.first(), will return the
-         * context (the this) and arguments for the first call
-         */
-        first(): CallInfo;
-        /**
-         * By chaining the spy with calls.reset(), will clears all tracking for a spy
-         */
-        reset(): void;
-    }
-
-    interface CallInfo {
-        /**
-         * The context (the this) for the call
-         */
-        object: any;
-        /**
-         * All arguments passed to the call
-         */
-        args: any[];
-        /**
-         * The return value of the call
-         */
-        returnValue: any;
-    }
-
-    interface CustomMatcherFactories {
-        [index: string]: CustomMatcherFactory;
-    }
-
-    type CustomMatcherFactory = (util: MatchersUtil, customEqualityTesters: CustomEqualityTester[]) => CustomMatcher;
-
-    interface MatchersUtil {
-        equals(a: any, b: any, customTesters?: CustomEqualityTester[]): boolean;
-        // tslint:disable-next-line: no-unnecessary-generics
-        contains<T>(haystack: ArrayLike<T> | string, needle: any, customTesters?: CustomEqualityTester[]): boolean;
-        buildFailureMessage(matcherName: string, isNot: boolean, actual: any, ...expected: any[]): string;
-    }
-
-    type CustomEqualityTester = (first: any, second: any) => boolean;
-
-    interface CustomMatcher {
-        compare<T>(actual: T, expected: T, ...args: any[]): CustomMatcherResult;
-        compare(actual: any, ...expected: any[]): CustomMatcherResult;
-    }
-
-    interface CustomMatcherResult {
-        pass: boolean;
-        message: string | (() => string);
-    }
-
-    interface ArrayLike<T> {
-        length: number;
-        [n: number]: T;
-    }
-}
+// Minimum TypeScript Version: 4.5
+
+declare var beforeAll: typeof import('@jest/globals').beforeAll;
+declare var beforeEach: typeof import('@jest/globals').beforeEach;
+declare var afterAll: typeof import('@jest/globals').afterAll;
+declare var afterEach: typeof import('@jest/globals').afterEach;
+declare var describe: typeof import('@jest/globals').describe;
+declare var fdescribe: typeof import('@jest/globals').fdescribe;
+declare var xdescribe: typeof import('@jest/globals').xdescribe;
+declare var it: typeof import('@jest/globals').it;
+declare var fit: typeof import('@jest/globals').fit;
+declare var xit: typeof import('@jest/globals').xit;
+declare var test: typeof import('@jest/globals').test;
+declare var xtest: typeof import('@jest/globals').xtest;
+declare var jest: typeof import('@jest/globals').jest;
```

Comment:
Does this re-export namespace stuff? That was an issue in https://github.com/facebook/jest/pull/12856/files#r945243675

Reply:
The `jest` namespace implementation could follow the code of 'packages/jest/src/index.ts' in the above mentioned PR.

Reply:
> Does this re-export namespace stuff?

It does not, but it could!

It’s just a bit annoying these exports need to be hand-picked.